sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -126,11 +126,11 @@ trait HashJoin {
           while (currentHashMatches == null && streamIter.hasNext) {
             currentStreamedRow = streamIter.next()
             val key = joinKeys(currentStreamedRow)
-            if (!key.anyNull) {
-              currentHashMatches = hashedRelation.get(key)
-              if (currentHashMatches != null) {
-                currentMatchPosition = 0
-              }
+            // We do filtering null keys by inserting a Filter before inner join,
```

Comment:
Even if we were going to make changes like this, they should be done in a different PR and come with some benchmarks to show that you are actually speeding things up.